Vector runtime panic when attempting aggregation on non-existent field

Repro is with super commit d53e041.

```
$ super -version
Version: v1.18.0-186-gd53e041a

$ echo '{"foo": "bar"}' | super -f parquet -o foo.parquet -

$ SUPER_VAM=1 super -c 'from "foo.parquet" | a:=avg(bar)'
panic: (*vector.Error) 0xc0005b0ee0

goroutine 72 [running]:
github.com/brimdata/super/runtime/vam/expr/agg.sumFlat[...](0xc000627818?, {0x3e3d030?, 0xc0005b0ee0?}, {0x0?, 0x0?, 0xc000692fc0?}, {0x0?, 0x221337f?, 0xc000627848?})
	/Users/phil/work/super/runtime/vam/expr/agg/mathfunc.go:69 +0x1a5
github.com/brimdata/super/runtime/vam/expr/agg.sum[...](0xc0005b0ee0?, {0x3e3d030?, 0xc0005b0ee0?})
	/Users/phil/work/super/runtime/vam/expr/agg/mathfunc.go:56 +0x14c
github.com/brimdata/super/runtime/vam/expr/agg.(*avg).Consume(0xc000909330, {0x3e3d030, 0xc0005b0ee0})
	/Users/phil/work/super/runtime/vam/expr/agg/avg.go:21 +0x67
github.com/brimdata/super/runtime/vam/op/summarize.(*superTable).update(0xc0005a8be0, {0xc000130070, 0x0, 0x1}, {0xc000130070, 0x1, 0xc000131f80?})
	/Users/phil/work/super/runtime/vam/op/summarize/agg.go:70 +0x659
github.com/brimdata/super/runtime/vam/op/summarize.(*Summarize).consume(0xc0004a5810, {0xc000130070, 0x0, 0x1}, {0xc000130070, 0x1, 0x1})
	/Users/phil/work/super/runtime/vam/op/summarize/summarize.go:102 +0x4cc
github.com/brimdata/super/runtime/vam/op/summarize.(*Summarize).Pull.func1({0xc000130070, 0x1, 0xc000627d80?})
	/Users/phil/work/super/runtime/vam/op/summarize/summarize.go:83 +0x78
github.com/brimdata/super/vector.Apply(0x0?, 0x3e3cff8?, {0xc000130070?, 0x0?, 0x3c39780?})
	/Users/phil/work/super/vector/apply.go:17 +0xa3
github.com/brimdata/super/runtime/vam/op/summarize.(*Summarize).Pull(0xc0004a5810, 0x0?)
	/Users/phil/work/super/runtime/vam/op/summarize/summarize.go:82 +0x13b
github.com/brimdata/super/runtime/vam/op.(*combineParent).run(0xc000910dc0)
	/Users/phil/work/super/runtime/vam/op/combine.go:110 +0x3f
created by github.com/brimdata/super/runtime/vam/op.(*Combine).Pull.func1 in goroutine 1
	/Users/phil/work/super/runtime/vam/op/combine.go:42 +0x35
```

It works if I run it in sequential runtime.

```
$ super -c 'from "foo.parquet" | a:=avg(bar)'
{a:null(float64)}
```

--- a/runtime/vam/expr/agg/avg.go
+++ b/runtime/vam/expr/agg/avg.go
@@ -14,6 +14,9 @@ type avg struct {
 var _ Func = (*avg)(nil)
 
 func (a *avg) Consume(vec vector.Any) {
+	if !super.IsNumber(vec.Type().ID()) {
+		return
+	}
 	var ncount uint32
 	if nulls := vector.NullsOf(vec); nulls != nil {
 		ncount = nulls.TrueCount()

